### DISCUSSION TITLE
How to start Directus 9.0.0-rc on Plesk 18.* with Nodejs plugin?

### DISCUSSION THREAD

**Discussion by 3logy** [directus/directus#3406] [Archive]
Hello everyone,

after installing directus with `yarn create directus-project my-project` in the document root of the subdomain, I want to start it using the Plesk Node.js tool. 
I tried differents settings, for example the last one see picture below, i ran `npm install`or `npx directus start` and unfortunately for each of them it doesnt works well for me. 

Has anyone figured out how to launch the application using the Plesk Nodejs plugin?

![grafik](https://user-images.githubusercontent.com/745075/102011347-c7419800-3d43-11eb-9851-c0bd1a21a5e0.png)

**Comment by rijkvanzanten** [comment]
I don't use Plesk myself, so haven't confirmed this, but `npx directus start` is the correct command. If it requires you to specify a file (I noticed that "application startup file" thingy), you might be able to use `node_modules/directus/dist/start.js`

**Reply by 3logy** [reply]
Thank you for the answer. I figure it out how to make it works.

**Reply by chinenvinicius** [reply]
hi i would like to know how u did it?

**Comment by 3logy** [comment]
Thank you for the answer. I figure it out how to make it works.

1. I installed, started nginx and add the following directives 

> location ~ / {
       proxy_redirect                      off;
       proxy_set_header Host               $http_host;
       proxy_set_header X-Real-IP          $remote_addr;
       proxy_set_header X-Forwarded-For    $proxy_add_x_forwarded_for;
       proxy_set_header X-Forwarded-Proto  $scheme;
       proxy_read_timeout          1m;
       proxy_connect_timeout       1m;
       proxy_pass                  http://localhost:8077; # 8077 is the port setup in the .env }

2. I setup the document root and application root to the original one
![grafik](https://user-images.githubusercontent.com/745075/102145346-57c4c900-3e67-11eb-91f1-746208f04322.png)
 
3. Then i add `"start": "npx directus start"` to the package.json (located in the root folder)
4. i then click on **npm install** and then **run script** : start

**Reply by ghost** [reply]
Did you add the Nginx Directives directly to the nginx.conf or to the vhost_nginx.conf?

**Comment by sapientorius** [comment]
Does someone know a more valid solution without starting a script manually?

It seems to work when you use the start.js file as "Application startup file", but only when not using port 80.

**Reply by rijkvanzanten** [reply]
> It seems to work when you use the start.js file as "Application startup file"

This sounds like a totally workable solution to me. The start.js file is what `npx directus start` calls as well 👍🏻

**Comment by lukas-schaetzle** [comment]
Also having trouble to get it to work. I could install all dependencies without an error but when I try to run `directus bootstrap` or `directus start` I get this error:
```
/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by /var/www/vhosts/<...>/node_modules/argon2/lib/binding/napi-v3/argon2.node)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! directus-test@1.0.0 start: `npx directus start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the directus-test@1.0.0 start script.
```

Any ideas on this one?

**Reply by lukas-schaetzle** [reply]
Ok, fixed it by rebuilding argon2 from source. :)

**Comment by lukas-schaetzle** [comment]
I finally managed to get it working. Using plesk for node.js projects is really underwhelming but unfortunately you are often forced to deal with it on cheap shared hosters. :/
Maybe it would be a good idea to add another installation guide for plesk? I can help out with a draft for it if you want.

**Reply by rijkvanzanten** [reply]
> Maybe it would be a good idea to add another installation guide for plesk?

Totally! I've never used it for node projects, and have no idea what it would entail, but i'll more than happily merge it into the docs if you have a draft 🙂

**Reply by Posoroko** [reply]
I tried the same thing, but in the node_modules folder of my project, I don't have a "directus" folder. So I can't find the start.js file... is that normal?
